lib/SparqlGenerator.js

```diff
@@ -226,28 +256,44 @@ var escape = /["\\\t\n\r\b\f]/g,
     escapeReplacements = { '\\': '\\\\', '"': '\\"', '\t': '\\t',
                            '\n': '\\n', '\r': '\\r', '\b': '\\b', '\f': '\\f' };
 
+Generator.prototype.toEntityIRI = function (iri) {
+  var longestPrefix, prefixLength = Number.NEGATIVE_INFINITY;
+  for (var prefix in this.prefixes) {
```

Comment:
Looping every time is expensive. Consider a technique such as https://github.com/RubenVerborgh/N3.js/blob/v0.9.1/lib/N3Writer.js#L225-L226